pkg/reconciler/ingress/ingress.go

```diff
@@ -424,3 +430,12 @@ func removeFinalizer(ingress *networkingv1.Ingress, finalizer string) {
 		}
 	}
 }
+
+func validateCustomHosts(ingress *networkingv1.Ingress, gblbDomain string) bool {
```

Comment:
```suggestion
func (c *Controller) validateCustomHosts(ingress *networkingv1.Ingress) bool {
        if *c.customHosts {
                return true
        }
        
        glbDomain := *c.domain
```

Currently this function is _checking if_ there are custom hosts rather than _validating_ the Ingress. I would add this bit of logic in this function and remove the `!*c.customHosts` clause in the call above to make this function actually validate the Ingress. Then the call to the function would be

```go
if !c.validateCustomHosts(ingress) {
```

Reply:
The reason I didn't add the function to the scope of the controller is that it doesn't change the scope of the controller and only checks if exists custom hosts in the ingress, it could well be a helper function that loops through an array and looks for existing elements.

Now that we are not preventing the ingress from being created I add the replaceCustomHosts that is part of the controller scope and is called if the `-enable-custom-hosts` flag is not set